docs/decisions/0002-mark-fuzzy-entries.rst

```diff
@@ -0,0 +1,82 @@
+Mark Transifex invalid entries as fuzzy
+#######################################
+
+Problem
+*******
+
+As part of the `OEP-58`_ Transifex GitHub App bot has now syncs Translations.
+Bot creates pull requests with invalid entries in the
+po-files. These entries are validated by `LexiQA`_, Transifex-builtin quality
+checks as well as human reviewers. However, some invalid translations still
+make it into the sync translation pull requests which breaks
+``django compilemessages`` command and is catchable by ``msgfmt``.
+
+If Transifex had marked the invalid entries as fuzzy, then the
+``django compilemessages`` command would have ignored them. However, that's
+not the case. Therefore, we need to perform this task ourselves.
+
+
+Design Decision
+***************
+
+Implement a GitHub Actions workflow that will mark the invalid entries as
+``fuzzy``. Then create a commit -- editing the pull request -- to branch
+of the bot pull request.
+
+To ensure a safe and reliable workflow, the following workflows will be
+combined into one single workflow with multiple jobs:
+
+#. Run ``python-tests.yml`` to validate the python code
+#. Then, run ``validate-translation-files.yml`` which performs the following:
+
+   #. Validate the po-files using ``msgfmt``
+   #. Mark the invalid entries as fuzzy (writes to the pofile)
+   #. Validate the files again to ensure the files are became valid
+   #. Commit the fixes if any
+
+#. Mark invalid entires as fuzzy therefore excluded from .mo files as well as msgfmt
+#. Validate the files again to ensure the files are still valid
+#. Commit the fixes
+#. Run the lint commits
+#. Push to the branch
+#. Trigger auto merge
+
+**Pros**
+
+* There's no need for manual intervention, therefor it's fast and won't
+  create a backlog of pull requests.
+
+
+**Cons**
+
+* The workflow script runs and edits the pull request, which can be
+  confusing for the reviewers.
+* The workflow lacks a notification mechanism to notify the reviewers
+  that their translations are invalid.
```

Comment:
I discussed this with Sarina and I think we can address this fairly elegantly by allowing this action to create issues on https://github.com/openedx/wg-translations/issues

It might take a bit of work to ensure we don't spam that repo with issues. I'd be happy to discuss how this could be implemented with you!

Reply:
You're right, we should implement a notification mechanism.

I think the following has the _same_ cost and complexity, therefore we should implement the one with the best Translator Experience.

## Option 1: WG Translations project GitHub issues
Create a list of people/teams to notify when things fail. One way is to utilize a [CODEOWNER](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) file to know who to nofity for which resource/language pair. Often the file has flexible enough rules to specific files belongs to which contributor.

If the CODEOWNER file format turns to be difficult, we can implement a YAML file to provide an easier to parse file in a similar manner.

## Option 2: Slack comments in a new `#translations-wg-issues` channel 
Post to a message to a new Slack channel e.g. `#translations-wg-issues` to announce the issue.

The Slack message may include a handle to the Slack CODEOWNER

 with handles to assign Slack users handles and let the bot notify them.

## Option 3: Email the contributors with their Transifex emails
Emails are not public, but definitely one way to tackle the issue

## Option 4: Open Transifex issues for each entry, and mark them as "unreviewed"
Get the `msgid` of the `fuzzy` entry then look it up on Transifex and open an issue. This notifies Transifex admins with a direct link to the entry via email. However, those issues are not visible to translators as far as I know.


We can use one or more notification mechanism based on what we see fit.

Reply:
Since goal is to implement the solution that provides translators with the best experience, I think this is a good question to ask in the #wg-translations slack channel.